app/requirements/base_requirement.py

```diff
@@ -0,0 +1,36 @@
+
+class BaseRequirement:
+
+    def __init__(self, requirement_info):
+        self.enforcements = requirement_info['enforcements']
+
+    def check_source_target(self, source, target):
+        """
+        Give a source and target fact, return True if the source and target comply with the enforcement mechanism and
```

Comment:
instead of writing this out in the description field, utilize the doc string elements:

param source: a fact object
param target: a fact object
return: True if the source and target comply...